infrastructure/smart-contracts/contracts/AchievementsFactory.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol";
+import "./Achievements.sol";
+
+/**
+ * @title AchievementsFactory
+ * @dev A factory contract that produces ERC1155 token contracts.
+ */
+contract AchievementsFactory is Initializable, AccessControlUpgradeable {
+    
+    //constants
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE"); //0xa49807205ce4d355092ef5a8a18f56e8913cf4a201fbe287825b095693c21775
+    bytes32 public constant MINT_ROLE = keccak256("MINT_ROLE"); //0x154c00819833dac601ee5ddded6fda79d9d8b506b911b3dbd54cdb95fe6c3686
+
+    //state
+    uint256 public productionCount; //number of contracts produced by this factory
+    mapping(string => address) public contractsById; //gameId => tokenContract
+
```

Comment:
We will need some way of tracking contracts a user has tokens minted of or we will only be able to get this info from events / offchain.

Reply:
Same as above